src/main/java/com/faforever/client/chat/AbstractChatTabController.java

```diff
@@ -198,6 +209,9 @@ public AbstractChatTabController(ClanService clanService, WebViewConfigurer webV
     this.autoCompletionHelper = autoCompletionHelper;
     this.eventBus = eventBus;
     this.countryFlagService = countryFlagService;
+    this.replayService = replayService;
+
+    replayBaseUrl = String.format(clientProperties.getVault().getReplayDownloadUrlFormat(), "");
```

Comment:
This approach works as long as the ID is at the end of the URL, but it might not be. A safer implementation would be to convert the string format into a regular expression that allows extracting the ID, like so:

```
    String quotedReplayUrlFormat = Pattern.quote(clientProperties.getVault().getReplayDownloadUrlFormat());
    replayUrlPattern = Pattern.compile(quotedReplayUrlFormat.replace("%s", "(.*?)"));
```

then, below, you can do:

```

    Matcher replayUrlMatcher = replayUrlPattern.matcher(url);
    if(replayUrlMatcher.matches()) {
      String replayId = replayUrlMatcher.group(1);
      ...
    }
```